plugins/processors/k8sdecorator/stores/utils_test.go

```diff
@@ -5,16 +5,94 @@ package stores
 
 import (
 	"github.com/docker/docker/pkg/testutil/assert"
+	"strconv"
 	"testing"
+	"time"
 )
 
 func TestUtils_parseDeploymentFromReplicaSet(t *testing.T) {
-	assert.Equal(t, "", parseDeploymentFromReplicaSet("cloudwatch-agent"))
-	assert.Equal(t, "cloudwatch-agent", parseDeploymentFromReplicaSet("cloudwatch-agent-42kcz"))
+	testcases := []struct {
+		name         string
+		inputString  string
+		expected     string
+	}{
+		{
+			name: "Get ReplicaSet Name with unallowed characters",
+			inputString: "cloudwatch-ag",
+			expected: "",
+		},
+		{
+			name: "Get ReplicaSet Name with allowed characters",
+			inputString: "cloudwatch-agent-42kcz",
+			expected: "cloudwatch-agent",
+		},
+		{
+			name: "Get ReplicaSet Name with string smaller than 3 characters",
+			inputString: "cloudwatch-agent-sd",
+			expected: "",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, parseDeploymentFromReplicaSet(tc.inputString), tc.expected, )
+		})
+	}
 }
 
 func TestUtils_parseCronJobFromJob(t *testing.T) {
-	assert.Equal(t, "", parseCronJobFromJob("hello-123"))
-	assert.Equal(t, "hello", parseCronJobFromJob("hello-1234567890"))
-	assert.Equal(t, "", parseCronJobFromJob("hello-123456789a"))
+	testcases := []struct {
+		name         string
+		inputString  string
+		expected     string
+	}{
+		{
+			name: "Get CronJobControllerV2 or CronJob's Name with alphabet characters",
+			inputString: "hello-name",
+			expected: "",
+		},
+		{
+			name: "Get CronJobControllerV2 or CronJob's Name with special characters and exact 10 characters",
+			inputString: "hello-1678995&64",
+			expected: "",
+		},
+		{
+			name: "Get CronJobControllerV2 or CronJob's Name with Unix Time not equal to 10 letters",
+			inputString: "hello-238",
+			expected: "",
+		},
+		{
+			name: "Get CronJobControllerV2's Name after k8s v1.21 with correct Unix Time",
+			inputString: "hello-"+strconv.FormatInt(time.Now().Unix()/60, 10),
```

Comment:
Nitpick - we only really need to get `time.Now()` once. The actual value isn't super important in these tests. We don't need to check the exact time multiple times. We only need one valid Unix timestamp that we can then call `strconv.FormatInt()` on it.

So something like:
```go
now := time.Now().Unix()
unixTime := strconv.FormatInt(now, 10)
unixMinutes := strconv.FormatInt(now/60, 10)

inputString1 := "hello-"+unixTime
inputString2 := "hello-"+unixMinutes+"foo"
```

Reply:
Reasonable. Fixed it in the newest commit.